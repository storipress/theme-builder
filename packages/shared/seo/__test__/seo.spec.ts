import { resolveSEOFromRaw } from '../index'

describe('resolveSEOFromRaw', () => {
  it('resolve seo with sepcial character', () => {
    expect(
      resolveSEOFromRaw({
        name: 'EOTC',
        base: {
          title: 'Beijing 2022 Closing Ceremony Recap - 再见 Beijing, Ciao Milano Cortina!',
          description:
            'The Beijing 2022 Winter Olympic Games has come to an official end with the Closing Ceremony, with all eyes now on Milano Cortina 2026!',
        },
        seo: JSON.parse(
          '{"meta":{"title":null,"matched":false,"description":"The Beijing 2022 Winter Olympic Games has come to an official end with the Closing Ceremony, with all eyes now on Milano Cortina 2026!"},"slug":null}'
        ),

        defaultSlug: 'beijing-2022-closing-ceremony-recap-再见-beijing-ciao-milano-cortina-ha5u',
      })
    ).toMatchSnapshot()
  })
})
