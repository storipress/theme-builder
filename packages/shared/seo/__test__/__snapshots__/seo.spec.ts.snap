// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`resolveSEOFromRaw > resolve seo with sepcial character 1`] = `
{
  "inject": {
    "footer": "",
    "header": "",
  },
  "meta": {
    "description": "The Beijing 2022 Winter Olympic Games has come to an official end with the Closing Ceremony, with all eyes now on Milano Cortina 2026!",
    "title": "Beijing 2022 Closing Ceremony Recap - 再见 Beijing, Ciao Milano Cortina! - EOTC",
  },
  "og": {
    "description": "The Beijing 2022 Winter Olympic Games has come to an official end with the Closing Ceremony, with all eyes now on Milano Cortina 2026!",
    "title": "Beijing 2022 Closing Ceremony Recap - 再见 Beijing, Ciao Milano Cortina! - EOTC",
  },
  "ogImage": "",
  "slug": "beijing-2022-closing-ceremony-recap-再见-beijing-ciao-milano-cortina-ha5u",
}
`;
